cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -135,6 +136,9 @@ where
 		let _final_extrinsics =
 			shuffle_extrinsics::<<Block as BlockT>::Extrinsic>(extrinsics, shuffling_seed);
 
+		// Reset the signal to include the inherent extrinsics on next `produce_bundle`.
+		self.to_include_inherents.store(true, Ordering::Relaxed);
```

Comment:
It is very non-intuitive that this line is needed here, I don't see how it is directly related to `produce_bundle` in this context and it will be very easy to forget in accidental refactoring.

Can we have an explicit argument to `produce_bundle` instead?

Reply:
I don't follow why an explicit argument to `produce_bundle`, it's actually a shared state between the `produce_bundle` and `process_bundles`. Could you elaborate?

Reply:
This API is magical. You need to call correct method from correct place and there is no obvious way to correlate one with another unless you have the graph of the implementation in your head. This is very error-prone.

Instead of `produce_bundle` reading boolean that is set by some magic code somewhere, it would be nicer to have an explicit argument and a clear data flow.

Reply:
I can get what you want, just don't get how to achieve that. It's an internal state maintained by the `Executor` instance, not a flag that should be set by the outside caller.

Reply:
Extract an inner implementation `produce_bundle_impl` and pass it as an explicit argument there, that's what I can come up with.

Reply:
That'll definitely be better IMHO

Reply:
Please see if https://github.com/subspace/subspace/pull/228/commits/9f2fc5061eb739ccde844dc757fd7eedd53a5abd is what you expect.

Reply:
Yes, except impl is moved into separate file again, making harder to discover and read :disappointed: